tests/unit/hooks/test_snapper.py

```diff
@@ -0,0 +1,265 @@
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar/.snapshots/1337/snapshot", "/baz"],
+        ),
+        (
+            {"include": ["/foo", "/foo", "/foo"], "exclude": ["/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar", "/baz"],
+        ),
+        (
+            {"include": "all", "exclude": ["/foo", "/bar", "/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo", "/bar", "/baz"],
+        ),
+        (
+            {"include": "all"},
+            ["/foo", "/bar", "/baz", "/quaz"],
+            [
+                "/foo/.snapshots/17/snapshot",
+                "/bar/.snapshots/1337/snapshot",
+                "/baz/.snapshots/13337/snapshot",
+                "/quaz",
+            ],
+        ),
+    ),
+)
+def test_use_snapshots_include_exclude(config, src_dirs_in, expected):
+    log_prefix = ''
+
+    flexmock(module).should_receive('execute_command_and_capture_output').with_args(
+        ["snapper", "--jsonout", "list-configs"]
+    ).and_return(
+        json.dumps(
+            {
+                "configs": [
+                    {
+                        "config": "foo",
+                        "subvolume": "/foo",
+                    },
+                    {
+                        "config": "bar",
+                        "subvolume": "/bar",
+                    },
+                    {
+                        "config": "baz",
+                        "subvolume": "/baz",
+                    },
+                ],
+            }
+        )
+    )
+    flexmock(module).should_receive('execute_command_and_capture_output').with_args(
+        ["snapper", "--jsonout", "-c", "foo", "list", "--disable-used-space"]
+    ).and_return(
+        json.dumps(
+            {
+                "foo": [
+                    {
+                        "number": 17,
+                    },
+                ]
+            }
+        )
+    )
+    flexmock(module).should_receive('execute_command_and_capture_output').with_args(
+        ["snapper", "--jsonout", "-c", "bar", "list", "--disable-used-space"]
+    ).and_return(
+        json.dumps(
+            {
+                "bar": [
+                    {
+                        "number": 1337,
+                    },
+                ]
+            }
+        )
+    )
+    flexmock(module).should_receive('execute_command_and_capture_output').with_args(
+        ["snapper", "--jsonout", "-c", "baz", "list", "--disable-used-space"]
+    ).and_return(
+        json.dumps(
+            {
+                "baz": [
+                    {
+                        "number": 13337,
+                    },
+                ]
+            }
+        )
+    )
+    flexmock(module.Path).should_receive("exists").and_return(True)
+
+    src_dirs_after = module.prepare_source_directories(config, log_prefix, src_dirs_in)
+    assert set(src_dirs_after) == set(expected)
```

Comment:
Also a nice-looking test! Seems to cover the bulk of the possible inputs/states.